cpg-neo4j/src/main/kotlin/de/fraunhofer/aisec/cpg/persistence/Common.kt

```diff
@@ -0,0 +1,244 @@
+ *                   \$$$$$   |$$ |      \$$$$$   |
+ *                    \______/ \__|       \______/
+ *
+ */
+package de.fraunhofer.aisec.cpg.persistence
+
+import de.fraunhofer.aisec.cpg.TranslationConfiguration
+import de.fraunhofer.aisec.cpg.TranslationContext
+import de.fraunhofer.aisec.cpg.graph.Name
+import de.fraunhofer.aisec.cpg.graph.Node
+import de.fraunhofer.aisec.cpg.graph.Persistable
+import de.fraunhofer.aisec.cpg.graph.edges.flows.DependenceType
+import de.fraunhofer.aisec.cpg.graph.edges.flows.Granularity
+import de.fraunhofer.aisec.cpg.helpers.BenchmarkResults
+import de.fraunhofer.aisec.cpg.helpers.neo4j.DataflowGranularityConverter
+import de.fraunhofer.aisec.cpg.helpers.neo4j.LocationConverter
+import de.fraunhofer.aisec.cpg.helpers.neo4j.NameConverter
+import de.fraunhofer.aisec.cpg.helpers.neo4j.SimpleNameConverter
+import de.fraunhofer.aisec.cpg.sarif.PhysicalLocation
+import java.math.BigInteger
+import kotlin.reflect.KClass
+import kotlin.reflect.KProperty1
+import kotlin.reflect.KTypeProjection
+import kotlin.reflect.KVisibility
+import kotlin.reflect.full.createType
+import kotlin.reflect.full.isSubtypeOf
+import kotlin.reflect.full.memberProperties
+import kotlin.reflect.full.superclasses
+import kotlin.reflect.full.withNullability
+import kotlin.reflect.jvm.javaType
+import kotlin.uuid.Uuid
+import org.neo4j.ogm.typeconversion.CompositeAttributeConverter
+import org.slf4j.LoggerFactory
+
+/**
+ * A cache used to store and retrieve sets of labels associated with specific Kotlin class types.
+ *
+ * This mutable map uses a Kotlin class type as the key and a set of strings representing associated
+ * labels as the value. The [labelCache] provides efficient lookup and prevents redundant
+ * re-computation of labels for the same class type.
+ */
+val labelCache: MutableMap<KClass<*>, Set<String>> = mutableMapOf()
+
+/**
+ * A cache mapping classes of type [Persistable] to their respective properties.
+ *
+ * This mutable map stores metadata about [Persistable] objects. For each specific class that
+ * implements the [Persistable] interface, it caches a mapping between property names and their
+ * corresponding [KProperty1] references. This allows efficient reflection-based access to class
+ * properties without repeatedly inspecting the class at runtime.
+ *
+ * The key in the map is the [KClass] of a subclass of [Persistable]. The value is a [Map] where the
+ * keys are strings representing the property names, and the values are [KProperty1] references
+ * pointing to those properties. This can be used for dynamic property access or serialization
+ * processes.
+ */
+val schemaPropertiesCache:
+    MutableMap<KClass<out Persistable>, Map<String, KProperty1<out Persistable, *>>> =
+    mutableMapOf()
+
+/**
+ * A set containing predefined property types represented as Kotlin type objects that can be used as
+ * properties in [schemaProperties].
+ */
+val propertyTypes =
+    setOf(
```

Comment:
Here we again have the issue that only hardcoded properties are persisted.

Reply:
Is propertyType even used?

Reply:
Removed